src/integrations/prefect-kubernetes/prefect_kubernetes/worker.py

```diff
@@ -166,12 +169,69 @@
 else:
     kubernetes = lazy_import("kubernetes")
 
+_LOCK = Lock()
+
 MAX_ATTEMPTS = 3
 RETRY_MIN_DELAY_SECONDS = 1
 RETRY_MIN_DELAY_JITTER_SECONDS = 0
 RETRY_MAX_DELAY_JITTER_SECONDS = 3
 
 
+class HashableKubernetesClusterConfig(BaseModel):
+    """
+    A hashable version of the KubernetesClusterConfig class.
+    Used for caching.
+    """
+
+    config: dict = Field(
+        default=..., description="The entire contents of a kubectl config file."
+    )
+    context_name: str = Field(
+        default=..., description="The name of the kubectl context to use."
+    )
+
+    def __hash__(self):
+        """Make the config hashable."""
+        return hash(
+            (
+                hash_objects(self.config),
+                self.context_name,
+            )
+        )
```

Comment:
there's a builtin setting if you're hashing all its fields
```suggestion
class HashableKubernetesClusterConfig(BaseModel, allow_mutation=False):
    """
    A hashable version of the KubernetesClusterConfig class.
    Used for caching.
    """

    config: dict = Field(
        default=..., description="The entire contents of a kubectl config file."
    )
    context_name: str = Field(
        default=..., description="The name of the kubectl context to use."
    )

Reply:
Oh nice. And we'll make it `frozen` in the 3.x version?